apps/api/src/app/events/services/storage-helper-service/storage-helper.service.ts

```diff
@@ -0,0 +1,56 @@
+import { Injectable } from '@nestjs/common';
+import { IAttachmentOptions } from '@novu/stateless';
+import { NonExistingFileError } from '../../../shared/errors/non-existing-file.error';
+import { StorageService } from '../../../shared/services/storage/storage.service';
+
+@Injectable()
+export class StorageHelperService {
+  constructor(private storageService: StorageService) {}
+
+  private areAttachmentsMissing(attachments?: IAttachmentOptions[]) {
+    return !(Array.isArray(attachments) && attachments.length > 0);
+  }
+
+  async uploadAttachments(attachments?: IAttachmentOptions[]) {
+    if (this.areAttachmentsMissing(attachments)) {
+      return;
+    }
+
+    const promises = attachments.map(async (attachment) => {
+      if (attachment.file) {
+        await this.storageService.uploadFile(attachment.name, attachment.file, attachment.mime);
```

Comment:
this section here is problematic due to `key` provided to the storage service. Imagine a scenario where the attachment name is "file.jpg", in our multi-tenant environment, if 2 people will send the same attachment name it will create a duplicate inside the folder. 

So the path we must use need to include the following information, both environment ID and organization id, and also the file name should be :
```
    const key = `${command.organizationId}/${command.environmentId}/${hat()}/${file.name}`;
```

This will also affect the getFile endpoint so that we need to make sure we persis the key somewhere on the attachments model so we can get them back later. 

@p-fernandez @davidsoderberg

Reply:
Yes sounds good 😄

Reply:
Good point, What is a `hat()` in the key @scopsy?
```
const key = `${command.organizationId}/${command.environmentId}/${hat()}/${file.name}`;
```

Reply:
@chavda-bhavik hat() is a function imported from [this](https://www.npmjs.com/package/hat) npm package 
it generates random id